pandas/core/internals/concat.py

```diff
@@ -260,6 +260,16 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
                     pass
                 elif getattr(self.block, "is_extension", False):
                     pass
+                elif is_extension_array_dtype(empty_dtype):
+                    missing_arr = empty_dtype.construct_array_type()._from_sequence(
+                        [], dtype=empty_dtype
+                    )
+                    ncols, nrows = self.shape
+                    assert ncols == 1, ncols
+                    empty_arr = -1 * np.ones((nrows,), dtype="int8")
```

Comment:
you actually want to use intp here

Reply:
Ah, OK, since that's what NumPy wants for indexing? For reference,

```python
In [2]: df1 = pd.DataFrame({"A": pd.array([], dtype='Int64')})

In [3]: df2 = pd.DataFrame({"B": np.arange(100000)})
```

```
# with intp
1.53 ms ± 10.5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

# with int8
1.73 ms ± 136 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```